pilot/pkg/xds/pcds.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	mesh "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	tb "istio.io/istio/pilot/pkg/trustbundle"
+	"istio.io/istio/pkg/util/gogo"
+)
+
+// PcdsGenerator generates ECDS configuration.
+type PcdsGenerator struct {
+	Server      *DiscoveryServer
+	TrustBundle *tb.TrustBundle
+}
+
+var _ model.XdsResourceGenerator = &PcdsGenerator{}
+
+func pcdsNeedsPush(req *model.PushRequest) bool {
+	if req == nil {
+		return true
+	}
+
+	if !req.Full {
+		// PCDS only handles full push
+		return false
+	}
+
+	// If none set, we will always push
+	if len(req.ConfigsUpdated) == 0 {
+		return true
+	}
+	return true
```

Comment:
Unsure if/how this can be further optimized. There are two parts to this 
(a) TrustAnchor updates should NOT result in push of other configuration (EDS etc) to proxies
(b) Other configuration changes (EDS etc) should not unnecessarily result in TrustAnchor update pushes

Reply:
You can return req.full at the moment, and add a comment to optimize.

Reply:
added a ProxyConfig type.